src/Angular-CSharp.csproj.in

```diff
@@ -20,7 +20,10 @@
     <PackageReference Include="Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore" Version="${MicrosoftAspNetCoreDiagnosticsEntityFrameworkCoreVersion}" Condition=" '$(IndividualLocalAuth)' == 'True' " />
     <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="${MicrosoftAspNetCoreIdentityEntityFrameworkCoreVersion}" Condition=" '$(IndividualLocalAuth)' == 'True' " />
     <PackageReference Include="Microsoft.AspNetCore.Identity.UI" Version="${MicrosoftAspNetCoreIdentityUIVersion}" Condition=" '$(IndividualLocalAuth)' == 'True' " />
-    <PackageReference Include="Microsoft.AspNetCore.SpaProxy" Version="${MicrosoftAspNetCoreSpaProxyVersion}" />
+    <PackageReference Include="Microsoft.AspNetCore.SpaProxy" Version="${MicrosoftAspNetCoreSpaProxyVersion}">
+      <!-- The SPA Proxy is only used at development time -->
+      <Publish>false</Publish>
```

Comment:
@amcasey is there a way we could do this inside the targets for the package?

I imagine this gets rid of the AoT problems we discussed in the past?

Reply:
@javiercn Sorry, I'm not sure I understand your suggestion.  I think you're hoping to make this implementation detail less visible, but I'm not sure how.

AFAICT, there were no AoT problems related to the SPA Proxy since it wasn't actually referenced anywhere - it was pulled in dynamically from a location specified in an environment variable.  The "issue" I was aware of was just the hygiene issue of marking all runtime assemblies as AOT-safe.

Reply:
> The "issue" I was aware of was just the hygiene issue of marking all runtime assemblies as AOT-safe.

Yes, this is what I meant. This makes the warnings go away, doesn't it?

I was saying that we ship a bunch of targets/props with the package [here](https://github.com/dotnet/aspnetcore/tree/main/src/Middleware/Spa/SpaProxy/src/build) and if there was an easy way of achieving the same result directly within the package.

I am not particularly concerned about the `<Publish>false</Publish>`, just wondering if we can make it less user facing.

Reply:
We hadn't actually enabled the analyzer, so there were no warnings, but this conclusively establishes that we don't need to enable it.

Oh, targets _in_ the SPA Proxy - well that makes more sense.  I have no idea if you can put something in the props/nuspec to prevent publication - seems a little weird that that wouldn't be up to the consumer, but I can check.

Reply:
I'm still having trouble understanding how this would work.  Maybe the targets could hook the publish target and remove the DLL if it's present?  That seems unnecessarily convoluted.  And it seems like any declarative solution would actually to go in the package metadata and not in the targets?  Nothing is impossible (especially since msbuild is Turing complete), but this is such a niche scenario - include in output but not in publish payload - that I'm skeptical such dedicated metadata would exist.  Do you have any pointers?

Reply:
@amcasey no, I was asking if you knew about a similar way to achieve it within the package. The current solution is completely reasonable to me, just wanted to confirm that there was no "easy" way to achieve this within the package itself.

Reply:
I'm not aware of one and my search of the docs didn't turn anything up.  I suspect I would have been steered towards one when I asked whether I could use an undocumented property, so I'm inclined to guess it's convoluted if it's possible at all.